packages/codecs-data-structures/src/__tests__/boolean-test.ts

```diff
@@ -1,36 +1,76 @@
-        expect(boolean().read(b('00'), 0)).toStrictEqual([false, 1]);
-        expect(boolean().read(b('ffff01'), 2)).toStrictEqual([true, 3]);
-        expect(boolean().read(b('ffff00'), 2)).toStrictEqual([false, 3]);
-        expect(boolean({ size: u32() }).read(b('01000000'), 0)).toStrictEqual([true, 4]);
-        expect(boolean({ size: u32() }).read(b('00000000'), 0)).toStrictEqual([false, 4]);
-
-        // Fails if the codec is not fixed size.
-        expect(() => boolean({ size: getShortU16Codec() })).toThrow(
-            new SolanaError(SOLANA_ERROR__CODECS__EXPECTED_FIXED_LENGTH),
-        );
-    });
-
-    it('has the right sizes', () => {
-        expect(boolean().fixedSize).toBe(1);
-        expect(boolean({ size: u32() }).fixedSize).toBe(4);
+    // A variable-size number codecs that uses 0 for `false`
+    // and the max shortU16 value for `true`.
+    const mappedShortU16 = mapCodec(
+        getShortU16Codec(),
+        // eslint-disable-next-line jest/no-conditional-in-test
+        v => (v === 0 ? 0 : 0xffff),
+        // eslint-disable-next-line jest/no-conditional-in-test
+        v => (v === 0 ? 0 : 1),
+    );
+
+    it('encodes booleans using a u8 number', () => {
+        expect(getBooleanCodec().encode(true)).toStrictEqual(b('01'));
+        expect(getBooleanCodec().encode(false)).toStrictEqual(b('00'));
+    });
+
+    it('decodes booleans using a u8 number', () => {
+        expect(getBooleanCodec().decode(b('01'))).toBe(true);
+        expect(getBooleanCodec().decode(b('00'))).toBe(false);
+    });
+
+    it('encodes booleans using a custom fixed-size number codec', () => {
+        const codec = getBooleanCodec({ size: getU32Codec() });
+        expect(codec.encode(true)).toStrictEqual(b('01000000'));
+        expect(codec.encode(false)).toStrictEqual(b('00000000'));
+    });
+
+    it('decodes booleans using a custom fixed-size number codec', () => {
+        const codec = getBooleanCodec({ size: getU32Codec() });
+        expect(codec.decode(b('01000000'))).toBe(true);
+        expect(codec.decode(b('00000000'))).toBe(false);
+    });
+
+    it('encodes booleans using a custom variable-size number codec', () => {
+        const codec = getBooleanCodec({ size: mappedShortU16 });
+        expect(codec.encode(true)).toStrictEqual(b('ffff03'));
+        expect(codec.encode(false)).toStrictEqual(b('00'));
+    });
+
+    it('decodes booleans using a custom variable-size number codec', () => {
+        const codec = getBooleanCodec({ size: mappedShortU16 });
+        expect(codec.decode(b('ffff03'))).toBe(true);
+        expect(codec.decode(b('00'))).toBe(false);
+    });
+
+    it('pushes the offset forward when writing', () => {
+        expect(getBooleanCodec().write(true, new Uint8Array(10), 6)).toBe(7);
+    });
+
+    it('pushes the offset forward when reading', () => {
+        expect(getBooleanCodec().read(b('ffff00'), 2)).toStrictEqual([false, 3]);
+    });
```

Comment:
Maybe not necessary, but you could do a few of these style tests with a custom boolean codec.

Reply:
Will do!